pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -392,6 +392,9 @@ private void validateSchemaName(Schema schema) {
       throw new ControllerApplicationException(LOGGER,
           "Invalid schema. Reason: 'schemaName' should not be null or empty", Response.Status.BAD_REQUEST);
     }
+    if (schema.getSchemaName().contains(TableConfig.TABLE_NAME_FORBIDDEN_SUBSTRING)) {
+      throw new IllegalStateException("'schemaName' cannot contain double underscore ('__')");
```

Comment:
Sorry merged it pre-maturely. Can you change this to a `ControllerApplicationException`?

Reply:
Yes, let me raise a PR.

Reply:
#14517